### DISCUSSION TITLE
delay ~18 secs before establishing connection

### DISCUSSION THREAD

**Discussion by inkcmd** [rustdesk/rustdesk#9373] [General]
Hello. we are using rustdesk server OSS, but i dont think that's OSS problem, but i don't have pro server to test it. i've got a delay ~18 secs between clicking "connect" and opening field "enter password" on win10 on both rustdesk versions x64 and x32, and on win7 x64, but x32 version connects instantly to any another computer, but we can't use forked adress book, and also we don't want to use win7, this is too old. maybe someone knows something about that? we use public keys on our server. settings are all the same, fresh installation, but x32 version of rustdesk looks a little bit different on windows. maybe someone have an advice, how to investigate it? rustdesk oss sever works perfectly and it's not a big issue, but there's no limit to perfection and of course we want to reduce this delay between clicking connect and entering the password to zero if possible, because there is no delay on win7 (x32 and x64) + x32 rustdesk client. we're using 1.3.0 rustdesk.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/discussions/7422
